internal/hcl/project_locator.go

```diff
@@ -232,18 +232,27 @@ func (p *ProjectLocator) walkPaths(fullPath string, level int) []string {
 	}
 
 	files := hclParser.Files()
-	var providerBlocks bool
+	var hasProviderBlock bool
+	var hasBackendAttr bool
 
 	for _, file := range files {
-		body, content, diags := file.Body.PartialContent(justProviderBlocks)
+		body, content, diags := file.Body.PartialContent(terraformAndProviderBlocks)
 		if diags != nil && diags.HasErrors() {
 			p.logger.Warn().Err(diags).Msgf("skipping building module information for file %s as failed to get partial body contents", file)
 			continue
 		}
 
-		// only do this after looping through all files
-		if len(body.Blocks) > 0 {
-			providerBlocks = true
+		providerBlocks := body.Blocks.OfType("provider")
+		if len(providerBlocks) > 0 {
+			hasProviderBlock = true
+		}
+
+		terraformBlocks := body.Blocks.OfType("terraform")
+		for _, block := range terraformBlocks {
+			backend, _, _ := block.Body.PartialContent(nestedBackendBlock)
+			if len(backend.Blocks) > 0 {
+				hasBackendAttr = true
```

Comment:
can be make sure we `break` here after finding a block so we don't need to do it more than required